Source/bmalloc/libpas/src/test/PGMTests.cpp

```diff
@@ -108,17 +109,52 @@ void testPGMMultipleAlloc() {
     const size_t num_allocations = 100;
     pas_allocation_result mem_storage[num_allocations];
 
-    for (size_t i = 0; i < num_allocations; i++ ) {
+    for (size_t i = 0; i < num_allocations; ++i) {
         size_t alloc_size = random() % 100000;
-        mem_storage[i] = pas_probabilistic_guard_malloc_allocate(&heap->large_heap, alloc_size, pas_non_compact_allocation_mode, &iso_heap_config, &transaction);
+        mem_storage[i] = pas_probabilistic_guard_malloc_allocate(&heap->large_heap, alloc_size, 1, pas_non_compact_allocation_mode, &iso_heap_config, &transaction);
         pas_allocation_result mem = mem_storage[i];
         memset(reinterpret_cast<void *>(mem.begin), 0x42, alloc_size);
     }
 
-    for (size_t i = 0; i < num_allocations; i++ ) {
+    for (size_t i = 0; i < num_allocations; ++i)
         pas_probabilistic_guard_malloc_deallocate(reinterpret_cast<void *>(mem_storage[i].begin));
+
+    size_t updated_free_virtual_mem = pas_probabilistic_guard_malloc_get_free_virtual_memory();
+    size_t updated_free_wasted_mem = pas_probabilistic_guard_malloc_get_free_wasted_memory();
+
+    CHECK_EQUAL(init_free_virtual_mem, updated_free_virtual_mem);
+    CHECK_EQUAL(init_free_wasted_mem, updated_free_wasted_mem);
+
+    pas_heap_lock_unlock();
+}
+
+void testPGMAlignedAlloc()
+{
+    pas_heap_ref heapRef = ISO_HEAP_REF_INITIALIZER_WITH_ALIGNMENT(getpagesize() * 100, 1);
+    pas_heap* heap = iso_heap_ref_get_heap(&heapRef);
+    pas_physical_memory_transaction transaction;
+    pas_physical_memory_transaction_construct(&transaction);
+
+    pas_heap_lock_lock();
+
+    size_t init_free_virtual_mem = pas_probabilistic_guard_malloc_get_free_virtual_memory();
+    size_t init_free_wasted_mem = pas_probabilistic_guard_malloc_get_free_wasted_memory();
+
+    const size_t num_allocations = 100;
+    pas_allocation_result mem_storage[num_allocations];
+
+    const size_t alignment = 32 << 10;
+    for (size_t i = 0; i < num_allocations; ++i) {
+        size_t alloc_size = random() % 100000;
+        pas_allocation_result mem = pas_probabilistic_guard_malloc_allocate(&heap->large_heap, alloc_size, alignment, pas_non_compact_allocation_mode, &iso_heap_config, &transaction);
+        mem_storage[i] = mem;
+        memset(reinterpret_cast<void *>(mem.begin), 0x42, alloc_size);
+        CHECK(pas_is_aligned(mem.begin, alignment));
     }
```

Comment:
We're only testing with 1 alignment value here.  Can we test this with different alignment values from 4K to 512K?  For the larger alignments, maybe we have to reduce `num_allocations`?